src/tools/pr_8799/src/virtcontainers.rs

```diff
@@ -0,0 +1,109 @@
+use anyhow::{anyhow, Result};
+use std::{collections::HashMap, fs, path::Path};
+
+// Create a map of tids-vcpus from /proc/pid/task subdirectories
+pub fn get_ch_vcpu_tids_by_path(proc_path: &str) -> Result<HashMap<u32, u32>> {
+    let mut tids = HashMap::new();
+
+    if proc_path.trim().is_empty() {
+        return Err(anyhow!("Path must be non-empty strings."));
+    }
+
+    let src = std::fs::canonicalize(proc_path)
+        .map_err(|e| anyhow!("Invalid /proc/pid path: {proc_path}: {e}"))?;
+
+    if !src.metadata()?.is_dir() {
+        let msg = format!("Invalid /proc/pid path: {}", src.display());
+        return Err(anyhow!(msg));
+    }
+
+    let tid_path = Path::new(&src).join("task");
+
+    // Make the map of TIDs/VCPUs
+    for entry in fs::read_dir(tid_path)? {
+        let entry = entry?;
+
+        let tid_str = match entry.file_name().into_string() {
+            Ok(id) => id,
+            Err(_) => continue,
+        };
+
+        let comm_path = Path::new("/proc").join(tid_str.clone()).join("comm");
```

Comment:
I don't think this is correct as we should be looking at `/proc/$cloud_hypervisor_pid/task/*/comm` only.